### PR TITLE
[JBJCA-1385] Keeping poolSize counter in sync in SemaphoreConcurrentL…

### PR DESCRIPTION
**Pull request by spyrkob**
…inkedDequeManagedConnectionPool

Issue: https://issues.jboss.org/browse/JBJCA-1385

### CODE REVIEW

**Comment by spyrkob** [comment]
@maeste could you please review? The issue is that a SemaphoreConcurrentLinkedDequeManagedConnectionPool#removeConnectionListenerFromPool can be called twice during a single getConnection() call (details in the linked JIRA) resulting in poolCounter getting out of sync with content of ConnectionListeners map.

If this look OK I'll try to add a unit test next week.

**Comment by spyrkob** [comment]
Unit test added

File: core/src/test/java/org/jboss/jca/core/connectionmanager/unit/pool/mcp/FailingManagedCF.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		ManagedConnection mc = (ManagedConnection) connectionSet.iterator().next();
+		SimpleConnection sc = (SimpleConnection) mc.getConnection(null, null);
+		sc.fail();
+		return connectionSet;
+	}
+
+	@Override
+	public ManagedConnection createManagedConnection(Subject subject, ConnectionRequestInfo cxRequestInfo)
+			throws ResourceException {
+		// TODO Auto-generated method stub
```
Comment (fl4via) [inline core/src/test/java/org/jboss/jca/core/connectionmanager/unit/pool/mcp/FailingManagedCF.java]: Please, remove this TODO

**Comment by fl4via** [comment]
Github is not showing it clearly, but I approved just the fix commit, the commit with the test needs just a small fix :-)

**Comment by spyrkob** [comment]
Sorry @fl4via missed that, removed TODO now. Thanks!

**Pull request merged**